packages/cli-hooks/src/protocols.js

```diff
@@ -0,0 +1,119 @@
+ *   Similar to stubbing functionality in mocking or testing setup utilties.
+ *   Ensures that the protocol expectations are met by userland and SDK code.
+ * @property {() => void} [uninstall] -
+ *   Optional instructions for the protocol to remove itself from the runtime.
+ */
+
+const DEFAULT_PROTOCOL = 'default';
+const MSG_BOUNDARY_PROTOCOL = 'message-boundaries';
+const SUPPORTED_NAMED_PROTOCOLS = [MSG_BOUNDARY_PROTOCOL];
+
+/**
+ * The baseline CLI-SDK protocol. All responses in this protocol go to stdout.
+ * The CLI combines both stdout and stderr to interpret the hook response.
+ * This simplistic protocol has inherent limitation: no logging diagnostics!
+ * @param {string[]} args - Command-line arguments passed to this process.
+ * @returns {Protocol} Specified communication rules for the SDK to follow.
+ */
+export function BaseProtocol(args) {
+  const { manifest: manifestOnly = false } = parseArgs(args);
+
+  // If the particular hook invocation is requesting manifest generation we
+  // ensure any logging is a no-op to prevent littering stdout with logging
+  // and confusing the CLI's manifest JSON payload parsing.
+  const loggerMethod = manifestOnly ? () => { } : console.log; // eslint-disable-line no-console
+  return {
+    name: DEFAULT_PROTOCOL,
+    log: loggerMethod,
+    error: loggerMethod,
+    warn: loggerMethod,
+    respond: console.log, // eslint-disable-line no-console
+  };
+}
+
+/**
+ * Protocol implementation that uses both stdout and stderr for logs and alerts,
+ * but also uses message boundaries to differentiate between application created
+ * diagnostic information and hook responses.
+ * @param {string[]} args - Command-line arguments passed to this process.
+ * @returns {Protocol} Specified communication rules for the SDK to follow.
+ */
+export function MessageBoundaryProtocol(args) {
+  const { boundary } = parseArgs(args);
+  if (!boundary) throw new Error('no boundary argument provided!');
+  const protocol = {
+    name: MSG_BOUNDARY_PROTOCOL,
+    log: console.log, // eslint-disable-line no-console
+    error: console.error, // eslint-disable-line no-console
+    warn: console.warn, // eslint-disable-line no-console
+    respond: (/** @type {any} */ data) => {
+      console.log(boundary + data + boundary); // eslint-disable-line no-console
+    },
+    getCLIFlags: () => [
+      `--protocol=${MSG_BOUNDARY_PROTOCOL}`,
+      `--boundary=${boundary}`,
+    ],
+  };
+  return protocol;
+}
+
+// A map of protocol names to protocol implementations
+const PROTOCOL_MAP = {
+  [SUPPORTED_NAMED_PROTOCOLS[0]]: MessageBoundaryProtocol,
+};
+
+/**
+ * Determines the protocol interface to use when communicating with the CLI.
+ * Based on the arguments provided by the CLI to the SDK hook process.
+ * @param {string[]} args - Command-line flags and arguments passed to the hook.
+ * @returns {Protocol} The interface to follow when messaging to the CLI.
+ */
+export function getProtocolInterface(args) {
```

Comment:
Nit: could we name this `getProtocol` since this function returns an implementation of a protocol